kt-generator/src/helpers.ts

```diff
@@ -0,0 +1,213 @@
+        case "LatLngPrimitiveType":
+            return "Location";
+        case "VoidPrimitiveType":
+            return "Unit";
+        case "AnyPrimitiveType":
+            return "Any";
+        case "OptionalType":
+            return `${generateTypeName((type as OptionalType).base)}?`;
+        case "ArrayType":
+            return `ArrayList<${generateTypeName((type as ArrayType).base)}>`;
+        case "StructType":
+            return type.name;
+        case "EnumType":
+            return type.name;
+        case "TypeReference":
+            return generateTypeName((type as TypeReference).type);
+        default:
+            throw new Error(`BUG: generateTypeName with ${type.constructor.name}`);
+    }
+}
+
+export function generateJsonAddRepresentation(type: Type, fieldName: string): string {
+    switch (type.constructor.name) {
+        case "StringPrimitiveType":
+        case "CpfPrimitiveType":
+        case "CnpjPrimitiveType":
+        case "EmailPrimitiveType":
+        case "PhonePrimitiveType":
+        case "CepPrimitiveType":
+        case "UrlPrimitiveType":
+        case "UuidPrimitiveType":
+        case "HexPrimitiveType":
+        case "Base64PrimitiveType":
+        case "SafeHtmlPrimitiveType":
+        case "XmlPrimitiveType":
+        case "IntPrimitiveType":
+        case "UIntPrimitiveType":
+        case "MoneyPrimitiveType":
+        case "FloatPrimitiveType":
+        case "BoolPrimitiveType":
+            return `addProperty(\"${fieldName}\", ${mangle(fieldName)})\n`;
+        case "OptionalType":
+            return generateJsonAddRepresentation((type as OptionalType).base, fieldName);
+        case "DatePrimitiveType":
+            return `addProperty(\"${fieldName}\", SimpleDateFormat(\"yyyy-MM-dd\", Locale.getDefault()).format(${mangle(fieldName)}))`;
+        case "DateTimePrimitiveType":
+            return `addProperty(\"${fieldName}\", SimpleDateFormat(\"yyyy-MM-dd'T'HH:mm:ss.SSS\", Locale.getDefault()).format(${mangle(fieldName)}))`;
+        case "LatLngPrimitiveType":
+        case "ArrayType":
+        case "StructType":
+        case "EnumType":
+        case "TypeReference":
+        case "AnyPrimitiveType":
+            return `add(\"${fieldName}\", gson.toJsonTree(${mangle(fieldName)}))`;
+        case "VoidPrimitiveType":
+            return "";
+        case "BytesPrimitiveType":
+            return `addProperty(\"${fieldName}\", android.util.Base64.encodeToString(${mangle(fieldName)}, android.util.Base64.DEFAULT))`;
+        default:
+            throw new Error(`BUG: No result found for generateJsonRepresentation with ${type.constructor.name}`);
+    }
+}
+
+export function mangle(fieldName: string): string {
+    const mangleList = [
+        "in",
+        "out",
+        "as",
+        "as?",
+        "break",
+        "class",
+        "continue",
+        "do",
+        "else",
+        "false",
+        "for",
+        "fun",
+        "if",
+        "in",
+        "!in",
```

Comment:
no need for this. "!in" is not a valid identifier anyway

Reply:
True, I just used a list of kotlin reserved words